src/main/scala/firrtl/ir/IR.scala

```diff
@@ -22,20 +22,45 @@ case object NoInfo extends Info {
   override def toString: String = ""
   def ++(that: Info): Info = that
 }
-case class FileInfo(info: StringLit) extends Info {
-  override def toString: String = " @[" + info.serialize + "]"
+
+/** Stores the string of a file info annotation in its escaped form. */
+case class FileInfo(escaped: String) extends Info {
+  override def toString: String = " @[" + escaped + "]"
   //scalastyle:off method.name
   def ++(that: Info): Info = if (that == NoInfo) this else MultiInfo(Seq(this, that))
+  def unescaped: String = FileInfo.unescape(escaped)
+}
+
+object FileInfo {
+  @deprecated("Use FileInfo.fromUnEscaped instead. FileInfo.apply will be removed in FIRRTL 1.5.", "FIRRTL 1.4")
+  def apply(info: StringLit): FileInfo = new FileInfo(escape(info.string))
+  def fromEscaped(s: String): FileInfo = new FileInfo(s)
+  def fromUnescaped(s: String): FileInfo = new FileInfo(escape(s))
+  /** prepends a `\` to: `\`, `\n`, `\t` and `]` */
+  def escape(s: String): String = {
+    s.replace("\\", "\\\\")
+      .replace("\n", "\\\n")
+      .replace("\t", "\\\t")
+      .replace("]", "\\]")
```

Comment:
This is broken, should replace `\n` with `\\n`!
And then we have a problem when trying to escape something like `"test\\\\t"` which should be escaped to `"test\\t"` not `"test\t"`

Reply:
TODO: replace with a custom Translator based on the apache commons: https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/StringEscapeUtils.java#L57

Reply:
The main difference should be that we do not escape unicode characters (or should we? at some point this should be documented in the Firrtl spec!) and `"` and that we do escape `]` because that is the delimiter used for the parser.

Reply:
If we can define that FIRRTL files are UTF-8 as you suggested (need to check that this is a good idea), then I think we should not escape unicode in emitted FIRRTL. But we certainly need it escaped in emitted Verilog.

Reply:
> But we certainly need it escaped in emitted Verilog.
Yes. I am working on moving the Verilog focused escaping into the Emitter (other parts of the compiler should not have to know about Verilog).

Reply:
Once I take more precise control over the escaping I can probably just add the unicode escaping on top of the Verilog stuff and unescape the `]` in one go.

Reply:
This is the newer version using `commons-text` instead of `commons-lang`: https://github.com/apache/commons-text/blob/master/src/main/java/org/apache/commons/text/StringEscapeUtils.java#L62